### PR TITLE
[Part-VIII] TF-2840 Calendar event reject interactor and controller

### PR DESCRIPTION
**Pull request by tddang-linagora**
### Issue
- #2840

### CODE REVIEW

File: lib/features/email/presentation/controller/single_email_controller.dart [hunk 1]
```diff
@@ -1715,6 +1722,18 @@ class SingleEmailController extends BaseController with AppLoaderMixin {
     }
   }
 
+  void _rejectCalendarEventAction() {
+    if (_rejectCalendarEventInteractor == null
+    || _displayingEventBlobId == null
+    || mailboxDashBoardController.accountId.value == null) {
```
Comment (dab246) [inline lib/features/email/presentation/controller/single_email_controller.dart:1728]: indent

File: lib/features/email/domain/usecases/calendar_event_reject_interactor.dart [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import 'package:core/core.dart';
```
Comment (dab246) [inline lib/features/email/domain/usecases/calendar_event_reject_interactor.dart:1]: only need import necessary class

File: lib/features/email/presentation/controller/single_email_controller.dart (hunk 1 above)
Comment (tddang-linagora) [reply lib/features/email/presentation/controller/single_email_controller.dart]: Done

File: lib/features/email/domain/usecases/calendar_event_reject_interactor.dart (hunk 2 above)
Comment (tddang-linagora) [reply lib/features/email/domain/usecases/calendar_event_reject_interactor.dart]: Done

File: lib/features/email/presentation/controller/single_email_controller.dart [hunk 3]
```diff
@@ -1734,6 +1753,14 @@ class SingleEmailController extends BaseController with AppLoaderMixin {
         appToast.showToastSuccessMessage(
           currentOverlayContext!,
           appLocalization.youWillAttendThisMeeting);
+      } else if (success is CalendarEventMaybeSuccess) {
+        appToast.showToastSuccessMessage(
+          currentOverlayContext!,
+          appLocalization.youMayAttendThisMeeting);
+      } else {
```
[the remark this answers is not in the document]
Comment (tddang-linagora) [reply lib/features/email/presentation/controller/single_email_controller.dart]: Done

**Review by dab246 — APPROVED** [review]
- [ ] Squash & rebase commit

**Review by dab246 — APPROVED** [review]
lgtm

**Review by hoangdat — APPROVED** [review]

**Pull request merged**